src/companies/models/material.py

```diff
@@ -0,0 +1,108 @@
+from datetime import timedelta
+from typing import Self
+
+from django.contrib.postgres.expressions import ArraySubquery
+from django.db import models
+from django.db.models import F, OuterRef, Q, QuerySet, Sum
+from django.db.models.functions import JSONObject, TruncDate
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+from app.models import DefaultModel
+from companies.models.stock import StockMaterial
+from purchases.models import UsedMaterial
+
+
+class MaterialType(DefaultModel):
+    name = models.CharField(
+        _("material type name"),
+        max_length=255,
+    )
+
+    class Meta:
+        ordering = ("name",)
+        verbose_name = _("material type")
+        verbose_name_plural = _("material types")
+
+
+class MaterialQuerySet(QuerySet):
+    def statistic(self, point_id: int, date_from: str | None = None, date_to: str | None = None) -> Self:
+        now = timezone.now().date()
+        q_date_from = Q(date__gte=date_from) if date_from else Q(date__gte=(now - timedelta(days=30)))
```

Comment:
There is a cleaner approach
```suggestion
        q_date_from = Q(date_from or (now - timedelta(days=30)))
```